acceptance/README-ec2.md

```diff
@@ -0,0 +1,85 @@
+# Testing on EC2
+
+## How To
+This section will provide step-by-step instructions for deploying
+an environment that will allow the `puppet` acceptance test suite
+to be executed on an EC2 master and agent.
+
+1. Make sure you have your AWS key and id defined in `$HOME/.fog`
+See the [beaker AWS documentation](https://github.com/puppetlabs/beaker/blob/master/docs/how_to/hypervisors/aws.md)
+for details.
+
+1. Clone puppet repo
+   ```
+   git clone git@github.com:puppetlabs/puppet.git
+   ```
+
+1. Perform a `bundle install` to install the gems required to run the tests.
+   ```
+   pushd puppet/acceptance
+   bundle install
+   ```
+
+1. Get the puppet [EC2 image templates](https://github.com/puppetlabs/pe_acceptance_tests/blob/2016.5.x/config/image_templates/ec2.yaml).
+   Note: These are restricted to Puppet, Inc. employees.
+   ```
+   pushd ../../
+   git clone git@github.com:puppetlabs/pe_acceptance_tests.git
+   pushd pe_acceptance_tests
+   git checkout 2017.2.x
+   popd
+   popd
+   mkdir config/image_templates
+   cp ../../pe_acceptance_tests/config/image_templates/ec2.yaml config/image_templates/
+   ```
+
+1. Set the value for the external `forge_host`
+
+   FIXME: The issues mentioned here have been resolved. Test that this is no
+   longer a blocker.
+
+   **BLOCKER**: Module tests relying on the test forge will be broken until
+   [BKR-1016](https://tickets.puppetlabs.com/browse/BKR-1016)
+   is resolved.
+
+   This is important. The hostname for the Puppet, Inc. test forge is different
+   depending on which side of Puppet, Inc. netork the test system is on. See [OPS-9499
+   resolution](https://tickets.puppetlabs.com/browse/OPS-9499?focusedCommentId=320139&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-320139)
+   for details.
+
+   For EC2 instances, it should be defined as follows:
+   ```
+   export forge_host=forgenext-aio-petest-dev-1.puppetlabs.net
+   ```
+
+   NOTE: Since the Rakefile sets a default value for `forge_host` at the options
+   level, this cannot be overridden by setting it at the host level via
+   beaker-hostgenerator. Currently, it needs to be set as an ENV var.
+
+1. Create the Beaker config file
+   Amazon Linux example
+   ```
+   bundle exec beaker-hostgenerator --global-config lifetime=2h "redhat6-64ma{hypervisor=ec2,user=ec2-user,amisize=c4.large,snapshot=pe,vmname=amazon-6-x86_64}-redhat6-64a{hypervisor=ec2,user=ec2-user,amisize=c4.large,snapshot=pe,vmname=amazon-6-x86_64}" > hosts.yaml
```

Comment:
This may be an issue outside the scope of this PR, but using `c4.large`, I was seeing this error while running the pre-suite:

```
aws-sdk: exception AWS::EC2::Errors::VPCResourceNotSpecified: The specified instance type can only be used in a VPC. A subnet ID or network interface ID is required to carry out the request.
```